docs/tests.md

```diff
@@ -0,0 +1,415 @@
+specifically based on the Python's doctest module.
+
+## Regression Tests: Understanding the Process
+
+At its core, our regression testing framework is designed to ensure that any
+changes made to the library don't inadvertently break existing functionality.
+Unlike doctests, which focus on individual components in isolation, regression
+tests take a broader approach by comparing the output of entire estimators
+against pre-computed expected results.
+
+This approach brings a significant advantage: **improved testing times.**
+Running each regression test, which often involves complex computations, can be
+very time-consuming. To optimize this process, we serialize expected outputs on
+predefined inputs. These serialized results are then used in subsequent test
+runs for efficient comparisons, saving valuable time during development.
+
+Here's a breakdown of how it works:
+
+```mermaid
+
+  flowchart LR;
+    gf("`Generators`")
+    inputs("`Inputs`");
+    preeo("`Precompute expected outputs`");
+    ser("`Serialize and store inputs/expected outputs`")
+    estimator("`Estimator`")
+    oc("`Test: Outputs comparison`")
+
+
+    gf --> preeo;
+    inputs --> preeo;
+    preeo --> ser;
+    ser -- input --> estimator;
+    estimator -- output --> oc;
+    ser -- expected output --> oc;
+
+```
+
+1. **Generating Expected Outputs:** For each estimator, we have specialized
+   generator functions that take defined input values and produce the expected
+   outputs. These generator functions are grouped by estimator and found in the
+   `tests/references/` directory. The generated outputs are then serialized to
+   be later used in the testing process.
+
+2. **Storing Test Inputs:** To maintain consistency, all input values used by
+   the generator functions are centrally defined in the `REFERENCES_INPUTS`
+   dictionary within `tests/references/references_inputs.py`. This ensures that
+   every test execution utilizes the same, controlled input data.
+
+3. **Executing Tests:** The actual test cases reside in the `tests/validations/`
+   directory, with each estimator having a dedicated file
+   (`test_<estimator_name>.py`). These test functions fetch input/output pairs
+   from a YAML file (which was pre-populated using the generator functions and
+   input data), execute the estimator's code, and compare the actual results
+   against the expected values.
+
+This separation of concerns makes it easier to maintain and extend the test
+suite as we introduce new estimators or modify existing ones.
+
+### Writing Tests: A Step-by-Step Guide
+
+This guide will walk you through the process of adding a new regression test to
+our Python library. We'll assume you're already familiar with Python, pytest,
+and the basic structure of our estimators.
+
+#### 1. Create Your Generator Function
+
+- Create a new Python file (`gen_<your_new_estimator>.py`) within the
+  `tests/references/<YourNewEstimator>Estimator/` directory.
```

Comment:
After each  `tests/references/<YourNewEstimator>Estimator/` it would be good to add an example maybe it the SDEstimator or the DummyEstimator